contracts/protocol/clients/FermionFractionsMint.sol

```diff
@@ -264,27 +298,55 @@ contract FermionFractionsMint is
         return IPriceOracleRegistry(fermionProtocol).isPriceOracleApproved(_oracle);
     }
 
-    // Overrides
     /**
-     * @notice Returns the number of fractions. Represents the ERC20 balanceOf method
+     * @notice Creates a new ERC20 clone for the current epoch.
      *
-     * @param _owner The address to check
+     * @param _epoch The epoch to create the clone for
+     * @return cloneAddress The address of the created clone
      */
-    function balanceOf(
-        address _owner
-    ) public view virtual override(ERC721, FermionFractionsERC20Base) returns (uint256) {
-        return FermionFractionsERC20Base.balanceOf(_owner);
+    function _createERC20Clone(uint256 _epoch) internal returns (address cloneAddress) {
+        FermionTypes.FermionFractionsStorage storage fractionStorage = Common._getFermionFractionsStorage();
+
+        cloneAddress = Clones.clone(erc20Implementation);
+        fractionStorage.epochToClone[fractionStorage.currentEpoch] = cloneAddress;
+
+        // Get the ERC721 storage directly
+        ERC721.ERC721Storage storage erc721Storage = Common._getERC721Storage();
+
+        // Format: name = "<fnft_name>_<epoch_index>", symbol = "<fnft_symbol><epoch_index>"
+        string memory name = string(abi.encodePacked(erc721Storage._name, "_", Strings.toString(_epoch)));
+        string memory symbol = string(abi.encodePacked(erc721Storage._symbol, Strings.toString(_epoch)));
+
+        FermionFractionsERC20(cloneAddress).initialize(name, symbol, address(this));
+
+        return cloneAddress;
     }
 
     /**
-     * @dev See {IERC20-transfer}.
-     *
-     * Requirements:
-     *
-     * - `to` cannot be the zero address.
-     * - the caller must have a balance of at least `value`.
+     * @dev Advances to the next epoch if the current epoch's total supply is 0.
+     * This function should be called when transitioning to a new epoch.
+     * Instead of just incrementing the epoch counter, it creates a new ERC20 clone for the new epoch.
+     * @return newEpoch The new epoch
      */
-    function transfer(address to, uint256 value) public virtual override(FermionFractionsERC20Base) returns (bool) {
-        return FermionFractionsERC20Base.transfer(to, value);
+    function _advanceEpoch() internal returns (uint256 newEpoch) {
+        FermionTypes.FermionFractionsStorage storage fractionStorage = Common._getFermionFractionsStorage();
+        uint256 currentEpoch = fractionStorage.currentEpoch;
+
+        // Check if we need to create the first ERC20 clone (for epoch 0)
+        if (currentEpoch == 0 && fractionStorage.epochToClone[0] == address(0)) {
+            newEpoch = 0;
+            fractionStorage.epochToClone[0] = _createERC20Clone(0);
```

Comment:
Small refactor is possible: 
You can put the line 347 (`fractionStorage.epochToClone[newEpoch] = _createERC20Clone(newEpoch);) out of the code block and remove the line 338. Also explicitly setting `newEpoch = 0` is not really necessary since it's the default value anyway.